plugins/hls-code-range-plugin/test/Ide/Plugin/CodeRangeTest.hs

```diff
@@ -50,5 +50,16 @@ testTree =
                         ( SelectionRange (Range (Position 1 1) (Position 5 10)) Nothing
                         )
                     )
+            ],
+        testGroup "findFoldingRanges" $
+            let check :: CodeRange -> [FoldingRange] -> Assertion
+                check codeRange = (findFoldingRanges codeRange @?=)
+
+                mkCodeRange :: Position -> Position -> V.Vector CodeRange -> CodeRange
+                mkCodeRange start end children = CodeRange (Range start end) children CodeKindRegion
+            in [
+                testCase "General" $ check
+                (mkCodeRange (Position 1 1) (Position 5 10) [])
+                [FoldingRange 1 (Just 1) 5 (Just 10) (Just FoldingRangeRegion)]
```

Comment:
@michaelpj @kokobd I was wondering what all test cases will be necessary to test folding ranges, or should I just add some general test cases of situations that happen(like switch cases, if else etc.)?

Reply:
Please add some test cases in `test/Main.hs` too. That file is for integration tests.

`Ide/Plugin/CodeRangeTest` contains unit tests. Your tests for `findFoldingRanges` is good ;) Maybe you could add a  test case for `createFoldingRange` too.